### PR TITLE
Implemented the 'resurrect' command handler

### PR DESCRIPTION
**Pull request by natrist**
See https://youtu.be/xRYPFcQEoxc for usage demonstration on WowGM

This change is [Reviewable](https://reviewable.io/reviews/mangostwo/server/175)

### CODE REVIEW

**Review by billy1arm — CHANGES_REQUESTED** [review]

File: src/game/Server/Opcodes.h [hunk 1]
```diff
@@ -1362,7 +1362,10 @@ enum Opcodes
     SMSG_COMMENTATOR_SKIRMISH_QUEUE_RESULT1         = 0x51C,// event EVENT_COMMENTATOR_SKIRMISH_QUEUE_REQUEST, CGCommentator::QueueNode
     SMSG_COMMENTATOR_SKIRMISH_QUEUE_RESULT2         = 0x51D,// event EVENT_COMMENTATOR_SKIRMISH_QUEUE_REQUEST
     SMSG_COMPRESSED_UNKNOWN_1310                    = 0x51E,// some compressed packet
-    NUM_MSG_TYPES                                   = 0x51F
+    SMSG_PLAYER_NOT_FOUND_FAILURE = 0x523,
+    SMSG_GM_RESURRECT_FAILURE = 0x524,
+    SMSG_GM_RESURRECT_SUCCESS = 0x528,
+    NUM_MSG_TYPES
```
[the remark this answers is not in the document]
Comment (natrist) [reply src/game/Server/Opcodes.h:1368]: This is intended behavior. See C++ reference for enum.

File: src/game/WorldHandlers/MiscHandler.cpp [hunk 2]
```diff
@@ -1309,6 +1309,57 @@ void WorldSession::WorldTeleportHandler(WorldPacket& recv_data)
...
+            if (pPlayer->IsAlive())
+            {
+                /* Player is already alive: Sending down failure response to client... */
+                DEBUG_LOG("resurrect failed");
+                SendGmResurrectFailureResponse();
+            }
+            else
+            {
+                /* Resurrect the player with full health and power... */
+				pPlayer->ResurrectPlayer(1.0f, false);
```
[the remark this answers is not in the document]
Comment (natrist) [reply src/game/WorldHandlers/MiscHandler.cpp:1341]: Good catch. I think it's the result of a bug with my Git GUI.
I'll be sending in a follow-up commit later today.

File: src/game/Server/Opcodes.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (billy1arm) [reply src/game/Server/Opcodes.h:1368]: Thanks for clarifying

**Review by natrist — COMMENTED** [review]
Everything looks good.

**Review by billy1arm — APPROVED** [review]

**Pull request merged**